ecs-init/config/common.go

```diff
@@ -99,3 +117,28 @@ func DockerUnixSocket() (string, bool) {
 	// eg: Docker daemon restart
 	return "/var/run", false
 }
+
+// ec2MetadataRegion finds the Region name from Metadata. If an error occurs fetching the region the default region name
+// is returned.
+func ec2MetadataRegion () string {
+	// Find Region name from Metadata. If error return a blank result
+	sessionInstance := session.Must(session.NewSession())
+	metadata := ec2metadata.New(sessionInstance)
+	regionName, err := metadata.Region()
+
+	if err != nil {
+		return DefaultRegionName
+	}
+
+	return regionName
+}
+
+// getS3BucketMapByRegion fetches the bucket URI from list of S3 Buckets by region name or default if key is not found
+func getS3BucketMapByRegion(regionName string) string {
```

Comment:
This should be named `getS3BucketForRegion` or (more generically) `getBaseLocationForRegion`.